en/components/spine.md

```diff
@@ -107,6 +107,166 @@ Vertex effect is only valid when the Animation Cache Mode property of Spine comp
+
+**Note**: After finish the Spine Attachments, you can delete unused skeleton nodes in the **Node Tree** panel to reduce the computational overhead at runtime. Note that the parent node of the target bone node cannot be deleted.
+
+### Implement in the script
+
+1. Similar to the steps of implement in the editor, first create a node with a Spine component and set the properties of the Spine component.
+
+2. Create the prefab of the star to be mounted on the skeleton animation. About Prefab please refer to [documentation](../asset-workflow/prefab.md) for details.
+
+3. Create a new JavaScript script in **Assets** and double-click to open to write. The sample script code is as follows:
+
+    ```js
+    cc.Class({
+        extends: cc.Component,
+
+        properties: {
+            skeleton: {
+                type: sp.Skeleton,
+                default: null,
+            },
+            // The prefab to be added to the skeleton animation
+            targetPrefab: {
+                type: cc.Prefab,
+                default: null,
+            },
+            // Target bone name
+            boneName: "",
+        },
+
+        onLoad () {
+            this.generateSomeNodes();
+        },
+
+        generateAllNodes () {
+            // Get the attach node tool
+            let attachUtil = this.skeleton.attachUtil;
+            attachUtil.generateAllAttachedNodes();
+            // Because there may be multiple skeletons with the same name, you need to return an array
+            let boneNodes = attachUtil.getAttachedNodes(this.boneName);
+            // Take the first skeleton as the attaching node
+            let boneNode = boneNodes[0];
+            boneNode.addChild(cc.instantiate(this.targetPrefab));
+        },
+
+        destroyAllNodes () {
+            let attachUtil = this.skeleton.attachUtil;
+            attachUtil.destroyAllAttachedNodes();
+        },
+
+        // A method to generate a tree of nodes with the specified skeleton name
+        generateSomeNodes () {
+            let attachUtil = this.skeleton.attachUtil;
+            let boneNodes = attachUtil.generateAttachedNodes(this.boneName);
+            let boneNode = boneNodes[0];
+            boneNode.addChild(cc.instantiate(this.targetPrefab));
+        },
+
+        // A method to destroy a tree of nodes with the specified skeleton name
+        destroySomeNodes () {
+            let attachUtil = this.skeleton.attachUtil;
+            attachUtil.destroyAttachedNodes(this.boneName);
+        }
+    });
+    ```
+
+4. Mount the script component onto the Canvas or other node, it means dragging and dropping the script into the **Properties** of the node. Then drag the nodes or resources to the corresponding properties box of script component, and save the scene.
```

Comment:
`**Properties**` -> `**Properties** panel`
Then drag the nodes or resources -> Then drag the node and resource